### ISSUE TITLE
Add check that MX point only to A/AAAA record in mailcow DNS Diagnostic Tool

### ISSUE THREAD

**Issue by dragoangel** [mailcow/mailcow-dockerized#3703]
**Is your feature request related to a problem? Please describe.**
Now DNS diagnostic tool doesn't notify administrator about incorrect MX pointing to CNAME record. Such solution is working, but isn't clearly not good: https://www.ietf.org/rfc/rfc1912.txt

Don't use CNAMEs in combination with RRs which point to other names like MX, CNAME, PTR and NS.  (PTR is an exception if you want to implement classless in-addr delegation.)  For example, this is strongly discouraged:

```
podunk.xx.      IN      MX      mailhost
mailhost        IN      CNAME   mary
mary            IN      A       1.2.3.4
```

[RFC 1034] in section 3.6.2 says this should not be done, and [RFC 974] explicitly states that MX records shall not point to an alias defined by a CNAME.  This results in unnecessary indirection in accessing the data, and DNS resolvers and servers need to work more to get the answer.

**Describe the solution you'd like**
Add warning if domain MX record is pointing to CNAME record

**Comment by andryyy** [comment]
Should be fixed with 1c8cca81879b5dee73313c8fb389095b638bbff2

**Comment by dragoangel** [comment]
Thanks Andre
